pkg/distributor/distributor.go

```diff
@@ -858,6 +859,100 @@ func (d *Distributor) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReq
 	return &mimirpb.WriteResponse{}, firstPartialErr
 }
 
+// StartBackfill requests the starting of a backfill session.
+func (d *Distributor) StartBackfill(ctx context.Context, tenantID int, blockID string) error {
+	level.Info(d.log).Log("msg", "starting backfill", "tenantId", tenantID, "blockId", blockID)
+	// TODO: Verify that block hasn't already been ingested
+	return nil
+}
+
+// AddBackfillFile adds a file to an ongoing metrics backfill.
+func (d *Distributor) AddBackfillFile(ctx context.Context, tenantID int, blockID, pth string, r *http.Request) error {
```

Comment:
[nit] I would rename "Add" to "Upload" to clarify the intention.

Reply:
Sounds good.

Reply:
Done.